python/lsst/ts/vent/controller/controller.py

```diff
@@ -0,0 +1,314 @@
+# This file is part of ts_atbuilding_vents
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import logging
+import os
+from enum import IntEnum
+
+import megaind
+import pymodbus.client
+
+from . import config as cfg
+from . import vfd
+
+__all__ = ["Controller"]
+logger = logging.getLogger(__name__)
+
+
+class VentGateState(IntEnum):
+    CLOSED = 0
+    """
+    The vent is closed
+    """
+
+    PARTIALLY_OPEN = 1
+    """
+    The vent is neither open nor closed
+    """
+
+    OPEN = 2
+    """
+    The vent is open
+    """
+
+    FAULT = -1
+    """
+    The vent is both open and closed
+    """
+
+
+class Controller:
+    """A controller that commands the components associated with the AT dome
+    vents and fans. The code in this class is meant to run on the Raspberry Pi
+    described in
+    https://confluence.lsstcorp.org/display/~fritzm/Auxtel+Vent+Gate+Automation
+    """
+
+    def __init__(self):
+        """Constructor."""
+        logger.debug("__init__")
+
+        vfd_hostname = cfg.VFD_HOSTNAME
```

Comment:
This comment is a bit of a pet-peeve on styling. See how you are kind of using cfg as a "class" here? This is the python way of doing a Singleton. 

I would suggest you convert the configuration into a data class or something. Then here you create an instance of that class owned by the `Controller`, or, better still, add a parameter to the class to receive the configuration data class. Then, you don't need to rely on environment variables for configuring these things.